marlowe-playground-client/src/AppM.purs

```diff
@@ -0,0 +1,42 @@
+module AppM where
+
+import Prelude
+import Control.Monad.Reader.Trans (class MonadAsk, ReaderT, ask, asks, runReaderT)
+import Effect.Aff (Aff)
+import Env (Env)
+import Effect.Class (class MonadEffect, liftEffect)
+import Effect.Aff.Class (class MonadAff, liftAff)
+import Type.Equality (class TypeEquals, from)
+
+newtype AppM a
+  = AppM (ReaderT Env Aff a)
+
+runAppM :: Env -> AppM ~> Aff
+runAppM env (AppM m) = runReaderT m env
+
+derive newtype instance functorAppM :: Functor AppM
+
+derive newtype instance applyAppM :: Apply AppM
+
+derive newtype instance applicativeAppM :: Applicative AppM
+
+derive newtype instance bindAppM :: Bind AppM
+
+derive newtype instance monadAppM :: Monad AppM
+
+derive newtype instance monadEffectAppM :: MonadEffect AppM
+
+derive newtype instance monadAffAppM :: MonadAff AppM
+
+-- | We can't write instances for type
+-- | synonyms, and we defined our environment (`Env`) as a type synonym for convenience. To get
+-- | around this, we can use `TypeEquals` to assert that types `a` and `b` are in fact the same.
+-- |
+-- | In our case, we'll write a `MonadAsk` (an alternate name for `Reader`) instance for the type
+-- | `e`, and assert it is our `Env` type. This is how we can write a type class instance for a
+-- | type synonym, which is otherwise disallowed.
```

Comment:
this is very cool, I didn't know you could do that. Also, `MonadAsk` is not the same as `MonadReader` so this comment is a bit misleading, see the definition of `MonadReader`

Reply:
Will change in the next PR because if I do it for this one I have to wait 3 hours before merging 😅.

I got the "hack" from the real halogen app https://github.com/thomashoneyman/purescript-halogen-realworld/blob/main/src/AppM.purs#L127